packages/runtime/src/runtimes/react/controls/rich-text-v2/EditableTextV2/editable-text-v2.tsx

```diff
@@ -131,9 +132,18 @@ export function EditableTextV2({ text, definition, control }: Props) {
   }, [editMode])
 
   useEffect(() => {
+    const defaultValue = definition.config.defaultValue ?? defaultText
+
     control?.setEditor(editor)
-    control?.setDefaultValue(defaultText)
-  }, [control, editor, defaultText])
+    control?.setDefaultValue(defaultValue)
+
+    /**
+     * When initialValue is set to the default value we need to trigger an local change so that the sidebar updates and so the data is saved
+     */
+    if (initialValue === defaultValue) {
+      control?.onLocalUserChange()
+    }
```

Comment:
The initialValue in the `<Editor />` is set without an onchange being fired so this makes sure the value is sent to the builder / the prop data is set.